module/friendship/app/query/list_friends.go

```diff
@@ -0,0 +1,75 @@
+package query
+
+import (
+	"context"
+
+	"github.com/phantranhieunhan/s3-assignment/common"
+	"github.com/phantranhieunhan/s3-assignment/common/logger"
+	"github.com/phantranhieunhan/s3-assignment/module/friendship/domain"
+)
+
+type ListFriendsRepo interface {
+	GetFriendshipByUserIDAndStatus(ctx context.Context, userID string, status ...domain.FriendshipStatus) (domain.Friendships, error)
+}
+
+type UserRepo interface {
+	GetUserIDsByEmails(ctx context.Context, emails []string) (map[string]string, error)
+	GetEmailsByUserIDs(ctx context.Context, userIDs []string) (map[string]string, error)
+}
+
+type ListFriendsHandler struct {
+	repo     ListFriendsRepo
+	userRepo UserRepo
+}
```

Comment:
`repository` and `handler` layer should be defined in separated folders and files

Reply:
Sorry, repo here, I mean that is the ports for each command or query conversion with the repo, I just renamed that for less confused, for example UserRepo => ListFriends_UserRepo. We do not reuse these interfaces for other commands or queries to be independent of them